modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/ddl/DdlSqlToCommandConverter.java

```diff
@@ -514,6 +539,59 @@ private CreateZoneCommand convertCreateZone(IgniteSqlCreateZone createZoneNode,
         return createZoneCmd;
     }
 
+
+    /**
+     * Converts the given IgniteSqlAlterZoneSet AST node to a AlterZoneCommand.
+     *
+     * @param alterZoneSet  Root node of the given AST.
+     * @param ctx        Planning context.
+     */
+    private DdlCommand convertAlterZoneSet(IgniteSqlAlterZoneSet alterZoneSet, PlanningContext ctx) {
+        AlterZoneSetCommand alterZoneCmd = new AlterZoneSetCommand();
+
+        alterZoneCmd.schemaName(deriveSchemaName(alterZoneSet.name(), ctx));
+        alterZoneCmd.zoneName(deriveObjectName(alterZoneSet.name(), ctx, "zoneName"));
+        alterZoneCmd.ifExists(alterZoneSet.ifExists());
+
+        Set<IgniteSqlZoneOptionEnum> knownOptionNames = EnumSet.allOf(IgniteSqlZoneOptionEnum.class);
+
+        for (SqlNode optionNode : alterZoneSet.alterOptionsList().getList()) {
+            IgniteSqlZoneOption option = (IgniteSqlZoneOption) optionNode;
+            IgniteSqlZoneOptionEnum optionName = option.key().symbolValue(IgniteSqlZoneOptionEnum.class);
+
+            if (!knownOptionNames.remove(optionName)) {
+                throw new IgniteException(QUERY_VALIDATION_ERR,
+                        String.format("Duplicate DDL command option has been specified [option=%s, query=%s]", optionName, ctx.query()));
+            }
+
+            DdlOptionInfo<AlterZoneSetCommand, ?> zoneOptionInfo = alterZoneOptionInfos.get(optionName);
+
+            assert zoneOptionInfo != null : optionName;
+            assert option.value() instanceof SqlLiteral : option.value();
+
+            updateCommandOption("Zone", optionName, (SqlLiteral) option.value(), zoneOptionInfo, ctx.query(), alterZoneCmd);
+        }
+
+        return alterZoneCmd;
+    }
+
+    /**
+     * Converts the given IgniteSqlAlterZoneRenameTo AST node to a AlterZoneCommand.
+     *
+     * @param alterZoneRename  Root node of the given AST.
+     * @param ctx        Planning context.
+     */
+    private DdlCommand convertAlterZoneRename(IgniteSqlAlterZoneRenameTo alterZoneRename, PlanningContext ctx) {
+        AlterZoneRenameCommand cmd = new AlterZoneRenameCommand();
+
+        cmd.schemaName(deriveSchemaName(alterZoneRename.name(), ctx));
+        cmd.zoneName(deriveObjectName(alterZoneRename.name(), ctx, "zoneName"));
+        cmd.newZoneName(deriveObjectName(alterZoneRename.newName(), ctx, "zoneName"));
```

Comment:
`deriveObjectName` is for identifiers which may be complex, but new name have to be simple, and we know this for sure. Probably, it would be better to call `alterZoneRename.newName().getSimple()` explicitly

Reply:
Fixed